src/peer_manager.rs

```diff
@@ -562,15 +564,15 @@ impl PeerManager {
     }
 
     /// Return the PeerIds of nodes bearing the names.
-    pub fn get_peer_ids(&self, names: &[XorName]) -> Vec<PeerId> {
+    pub fn get_peer_ids(&self, names: &HashSet<XorName>) -> Vec<PeerId> {
         names.iter()
             .filter_map(|name| self.peer_map.get_by_name(name).and_then(Peer::peer_id))
             .cloned()
             .collect()
     }
 
     /// Return the PublicIds of nodes bearing the names.
-    pub fn get_pub_ids(&self, names: &[XorName]) -> Vec<PublicId> {
+    pub fn get_pub_ids(&self, names: &HashSet<XorName>) -> HashSet<PublicId> {
```

Comment:
Well, it's probably not something I'd worry about too much until profiling, but checking in the `else` wouldn't be quite as efficient as calling `contains()` I guess - but clarity would trump performance for me, so yes, I'd say it's clearer to just do the additional check for our own name in the existing `filter_map()` run.

Anyway, I'll leave as is for now to allow us to progress this PR.